src/main/webapp/app/overview/submission-result-status.component.ts

```diff
@@ -46,9 +47,20 @@ export class SubmissionResultStatusComponent implements OnChanges {
             const quizExercise = this.exercise as QuizExercise;
             this.uninitialized = ArtemisQuizService.isUninitialized(quizExercise);
             this.quizNotStarted = ArtemisQuizService.notStarted(quizExercise);
+            this.shouldShowResult = !!this.studentParticipation?.results?.length;
         } else {
             this.uninitialized = !afterDueDate && !this.studentParticipation;
-            this.notSubmitted = !!this.studentParticipation && !this.studentParticipation.submissions?.length;
+            this.notSubmitted = afterDueDate && !!this.studentParticipation && !this.studentParticipation.submissions?.length;
+
+            if (this.exercise.type === ExerciseType.PROGRAMMING) {
+                const initializationStatesForResult = [InitializationState.INITIALIZED, InitializationState.INACTIVE, InitializationState.FINISHED];
```

Comment:
Does it make sense to move this to the top to not instantiate this array every time?

Reply:
That instantiation cost near to nothing and onChanges is also not to frequently called here. So performance wise it does not matter and an extra variable would not make the code cleaner

Reply:
It makes sense for code quality and readability point of view. The way it is done now anytime someone new looks at this code, he or she will think that there is something specific to this method runtime initialised here

Reply:
It's more about the general principle. If you have a static constant (compared to a constant that is filled with a variable input), you define it at the topmost level.

Reply:
Ok, I changed it for you